### PR TITLE
DlgPrefColors: use full palette preview for icons in comboboxes

### PR DESCRIPTION
**Pull request by Be-ing**
image
Screenshot from 2020-03-24 21-54-42

### CODE REVIEW

File: src/preferences/dialog/dlgprefcolors.cpp [hunk 1]
```diff
@@ -147,15 +150,15 @@ void DlgPrefColors::slotApply() {
 QPixmap DlgPrefColors::drawPalettePreview(const QString& paletteName) {
     ColorPalette palette = m_colorPaletteSettings.getHotcueColorPalette(paletteName);
     if (paletteName == palette.getName()) {
+        QPixmap pixmap(kPalettePreviewSize);
         int count = math_max(palette.size(), 1);
-        int width = math_min((200 / count), 16);
-        QPixmap pixmap(count * width, 16);
```
Comment (daschuer) [inline src/preferences/dialog/dlgprefcolors.cpp]: This code was to avoide a black border at the end of the strip. Now the border is back.
What was the reason to remove it?
Do we need an alternative solution?

File: src/preferences/dialog/dlgprefcolors.cpp [hunk 2]
```diff
@@ -180,14 +183,10 @@ QIcon DlgPrefColors::drawPaletteIcon(const QString& paletteName) {
 }
 
 void DlgPrefColors::slotTrackPaletteChanged(const QString& paletteName) {
```
Comment (daschuer) [inline src/preferences/dialog/dlgprefcolors.cpp]: We can now dispose the whole slot.

File: src/preferences/dialog/dlgprefcolorsdlg.ui [hunk 3]
```diff
@@ -26,36 +26,15 @@
...
-        </property>
-       </widget>
-      </item>
-      <item row="4" column="0">
-       <widget class="QLabel" name="labelHotcueDefaultColor">
-        <property name="text">
-         <string>Hotcue default color</string>
-        </property>
-       </widget>
+       <widget class="QComboBox" name="comboBoxHotcueDefaultColor"/>
```
Comment (daschuer) [inline src/preferences/dialog/dlgprefcolorsdlg.ui]: Please restore the expanding style.
This removes the gap between the labels an the box.

Comment (Be-ing) [reply src/preferences/dialog/dlgprefcolorsdlg.ui]: I don't know what you mean. All I did was delete widgets in Qt Creator and save. I don't see a difference in the spacing compared to the screenshots in https://github.com/mixxxdj/mixxx/pull/2589 If there's something you think needs fixing here, could you take care of it?

File: src/preferences/dialog/dlgprefcolors.cpp (hunk 2 above)
Comment (Be-ing) [reply src/preferences/dialog/dlgprefcolors.cpp]: okay

File: src/preferences/dialog/dlgprefcolors.cpp (hunk 1 above)
Comment (Be-ing) [reply src/preferences/dialog/dlgprefcolors.cpp]: The magic number 200 made no sense. I fixed the black edge in 85f1ce6.

**Comment by Be-ing** [comment]
Ready for merge?

**Comment by daschuer** [comment]
Thank you. LGTM.

**Pull request merged**